File: Assignment44/program4.c
//////////////////////////////////////////////////////////////////////
// File name :       program4.c
// Discription :     It is program which accept matrix and print 
//                   Addition of each columns.
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

//////////////////////////////////////////////////////////////////////
// Function Name :   AddColumn
// Discription :     It is used to print Addition of each columns.
// Input  :          int,int, int** 
// Output :          -
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////

void AddColumn(int rows, int cols,int **Arr)
{
    int iAdd = 0;

    
    for(int i = 0 ; i < rows ; i++)
    {
        for(int j = 0; j < cols; j++)
        {
            iAdd = iAdd + Arr[j][i];
        }
        printf("Addition of %d col : %d\n",(i+1),iAdd);
        iAdd = 0;
    }

} 

int main()
{
    int rows = 0,cols = 0,i = 0,j = 0;
    int iRet = 0,iValue = 0;

    printf("Enter the rows : \n");
    scanf("%d",&rows);

    printf("Enter the cols : \n");
    scanf("%d",&cols);
    
    int **Arr = malloc(rows * sizeof(int *));
    
    for(i = 0; i < rows; i++)
    {
        Arr[i] = malloc(cols * sizeof(int));
    }
    printf("Enter elements:\n");
    for(i = 0; i < rows; i++) {
        for(j = 0; j < cols; j++) {
            scanf("%d", &Arr[i][j]);
        }
    }
    printf("Elements are in Array : \n");

    for(i = 0; i < rows; i++)
    {
        for(j = 0; j < cols; j++)
        {
            printf("%d\t",Arr[i][j]);
        }
        printf("\n");
    }
    printf("\n");


    AddColumn(rows,cols,Arr);
  
    for(i = 0; i < rows; i++)
    {
        free(Arr[i]);
    }
    free(Arr);

    return 0;
}

File: Assignment51/program4.java
//////////////////////////////////////////////////////////////////////
// File name :       program4.java
// Discription :     It is program which accept matrix and calculate 
//                   trace and normal of matrix.
// Author :          Mangesh Ashok Bedre
// Date :            27/11/2025
//
//////////////////////////////////////////////////////////////////////

import java.util.Scanner;
import java.lang.Math;

class Matrix
{

    //////////////////////////////////////////////////////////////////////
    // Function Name :   CalTraceAndNormal
    // Discription :     It is used to calculate trace and normal of matrix.
    // Input  :          int[][] 
    // Output :          -
    // Author :          Mangesh Ashok Bedre
    // Date :            27/11/2025
    //
    //////////////////////////////////////////////////////////////////////  

    public float[] CalTraceAndNormal(int Arr[][])
    {
        int rows = Arr.length;
        int cols = Arr[0].length;
        float Result[] = new float[2];
        int iTrace = 0,iNormal = 0;

        for(int i = 0 ; i < rows ; i++)
        {
            for(int j = 0; j < cols; j++)
            {    
                if(i == j)
                {
                    iTrace = iTrace + Arr[i][j];
                }
                iNormal += Arr[i][j];
            }
        }
        Result[1] = (float)Math.sqrt(iNormal);
        Result[0] = iTrace;

        return Result;
    }
}

class program4
{
    public static void main(String a[])
    {

        int i = 0,j = 0;
        float fRet[];
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the rows : ");
        int rows = sobj.nextInt();

        System.out.println("Enter the cols : ");
        int cols = sobj.nextInt();

        int Arr[][] = new int [rows][cols];

        System.out.println("Enter the elements : ");
        for(i = 0; i < rows; i++) 
        {
            for(j = 0; j < cols; j++) 
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
        System.out.println("Elements are : ");
        for(i = 0; i < rows; i++) 
        {
            for(j = 0; j < cols; j++) 
            {
               System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
        

        Matrix mobj = new Matrix();
        fRet = mobj.CalTraceAndNormal(Arr);

        System.out.println("Trace of matrix : "+fRet[0]);
        System.out.println("Normal of matrix : "+fRet[1]);

    }
}